ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py

```diff
@@ -170,91 +162,92 @@ def _list_dirs_in_directory(pwd):
+
+    @staticmethod
+    @expose_to_cli('ovs', 'celery-ports-test')
+    def check_rabbitmq_ports(result_handler):
+        """
+        Checks all ports of Open vStorage components rabbitMQ and celery
 
+        :param result_handler: logging object
+        :type result_handler: ovs.extensions.healthcheck.result.HCResults
+        :return: True if successful; False it doesn't work
+        :rtype: bool
+        """
+        test_name = 'celery-ports-test'
         # Check Celery and RabbitMQ
-        logger.info('Checking RabbitMQ/Celery ...', '')
-        if Helper.get_ovs_type() == 'MASTER':
-            pcommand = 'celery inspect ping -b amqp://ovs:0penv5tor4ge@{0}//'.format(OpenvStorageHealthCheck.LOCAL_SR.ip)
-            pcel = commands.getoutput(pcommand.format(pcommand)).splitlines()
-            if len(pcel) != 1 and 'pong' in pcel[1].strip():
-                logger.success('Connection successfully established!', 'port_celery')
+        if Helper.get_ovs_type() != 'MASTER':
+            result_handler.skip('RabbitMQ is not running/active on this server!', 'port_celery')
+        result_handler.info('Checking Celery.')
+        from errno import errorcode
+        try:
+            from celery.task.control import inspect
+            stats = inspect().stats()
+            if stats:
+                result_handler.success('Successfully connected to Celery on all nodes.', test_name)
             else:
-                logger.failure('Connection FAILED to service Celery, please check RabbitMQ and ovs-workers?',
-                               'port_celery')
-        else:
-            logger.skip('RabbitMQ is not running/active on this server!', 'port_celery')
+                result_handler.failure('No running Celery workers were found.', test_name)
+        except IOError as ex:
+            msg = 'Could not connect to Celery. Got {0}.'.format(ex)
+            if len(ex.args) > 0 and errorcode.get(ex.args[0]) == 'ECONNREFUSED':
+                msg += ' Check that the RabbitMQ server is running.'
+                result_handler.failure(msg, test_name)
+        except ImportError as ex:
+            result_handler.failure('Could not import the celery module. Got {}'.format(str(ex)), test_name)
 
     @staticmethod
     @expose_to_cli('ovs', 'packages-test')
-    def check_ovs_packages(logger):
+    def check_ovs_packages(result_handler):
         """
         Checks the availability of packages for Open vStorage
 
-        :param logger: logging object
-        :type logger: ovs.log.healthcheck_logHandler.HCLogHandler
+        :param result_handler: logging object
+        :type result_handler: ovs.extensions.healthcheck.result.HCResults
         """
 
-        logger.info('Checking OVS packages: ', 'check_ovs_packages')
+        result_handler.info('Checking OVS packages: ', 'check_ovs_packages')
 
         for package in Helper.packages:
```

Comment:
I don't agree we check the packages only configured in the settings.json
but besides that i challenge you to add a non-existing package to that list and see how the healthcheck reacts